### PR TITLE
Giftcard: payment methods

### PR DESCRIPTION
**Pull request by jreij**
- Fetch order status from checkout shopper.
- Update payment methods screen.
- Update gift card payment confirmation screen.

### CODE REVIEW

File: drop-in/src/main/java/com/adyen/checkout/dropin/ui/paymentmethods/PaymentMethodAdapter.kt [hunk 1]
```diff
@@ -56,12 +59,22 @@ class PaymentMethodAdapter(
...
         }
     }
 
     private fun bindHeader(holder: HeaderVH, position: Int) {
         val header = getHeaderAt(position)
         holder.title.setText(header.titleResId)
+        if (header.actionResId == null) {
+            holder.action.visibility = View.GONE
+        } else {
+            holder.action.visibility = View.VISIBLE
```
Comment (ozgur00) [inline drop-in/src/main/java/com/adyen/checkout/dropin/ui/paymentmethods/PaymentMethodAdapter.kt:72]: when there's repetitive usage of same field you can use `with()` like:
```
with(holder.action) {
    visibility = View.VISIBLE
    setText(header.actionResId)
    setOnClickListener {
        onHeaderActionClick(header)
    }
}
```

File: drop-in/src/main/java/com/adyen/checkout/dropin/ui/DropInViewModel.kt [hunk 2]
```diff
@@ -74,7 +88,7 @@ class DropInViewModel(private val savedStateHandle: SavedStateHandle) : ViewMode
             savedStateHandle[IS_WAITING_FOR_RESULT_KEY] = value
         }
 
-    var cachedGiftCardComponentState: GiftCardComponentState?
+    private var cachedGiftCardComponentState: GiftCardComponentState?
```
Comment (ozgur00) [inline drop-in/src/main/java/com/adyen/checkout/dropin/ui/DropInViewModel.kt:91]: we never reset this. is there possibly a case where we need to reset this to `null`?

File: drop-in/src/main/java/com/adyen/checkout/dropin/ui/DropInActivityEvent.kt [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+ *
+ * Created by josephj on 29/11/2021.
+ */
+
+package com.adyen.checkout.dropin.ui
+
+import com.adyen.checkout.components.PaymentComponentState
+
+sealed class DropInActivityEvent {
+    class MakePartialPayment(val paymentComponentState: PaymentComponentState<*>) : DropInActivityEvent()
```
[the remark this answers is not in the document]
Comment (jreij) [reply drop-in/src/main/java/com/adyen/checkout/dropin/ui/DropInActivityEvent.kt:14]: I had no idea data classes could extend sealed classes!

File: drop-in/src/main/java/com/adyen/checkout/dropin/ui/DropInViewModel.kt (hunk 2 above)
Comment (jreij) [reply drop-in/src/main/java/com/adyen/checkout/dropin/ui/DropInViewModel.kt:91]: We don't need to reset now it but I don't like how it is now so I'm refactoring this part in the next PR

**Review by ozgur00 — APPROVED** [review]

**Pull request merged**